code/obj/flock/furniture.dm

```diff
@@ -211,6 +223,19 @@
 		<br><span class='bold'>System Integrity:</span> [round((src.health_attack/src.health_max)*100)]%
 		<br><span class='bold'>###=-</span></span>"}
 
+// flockdrones can always move through
+/obj/storage/closet/flock/Crossed(atom/movable/mover)
+	. = ..()
+	var/mob/living/critter/flock/drone/drone = mover
+	if(!src.open && istype(drone) && !drone.floorrunning)
+		animate_flock_passthrough(mover)
```

Comment:
```suggestion
	if(istype(drone) && !drone.floorrunning)
		if (!src.open)
			animate_flock_passthrough(mover)
```

Reply:
Or src.open check could just be placed earlier, as an early return type of thing

Reply:
literally the exact same behaviour, just slower. If `src.open` is `TRUE`, then the parent function will evaluate to `TRUE`. This change would just mean we did and extra `istype()` and a boolean check when the closet is already open that we don't need to do.

Reply:
also worth noting than in an if statement, byond does optimisations like in `if(false_func() & true_func())` because `false_func()` returns `FALSE`, `true_func()` is never called, meaning `if(!src.open & ...` is equivalent to an early return.